src/qt/rpcconsole.cpp

```diff
@@ -336,6 +344,54 @@ static QString categoryClass(int category)
     }
 }
 
+/** Restart wallet with "-salvagewallet" */
+void RPCConsole::wallet_salvage()
+{
+    build_parameter_list(QString("-salvagewallet"));
+}
+
+/** Restart wallet with "-rescan" */
+void RPCConsole::wallet_rescan()
+{
+    build_parameter_list(QString("-rescan"));
+}
+
+/** Restart wallet with "-zapwallettxes=1" */
+void RPCConsole::wallet_zaptxes1()
+{
+    build_parameter_list(QString("-zapwallettxes=1"));
+}
+
+/** Restart wallet with "-zapwallettxes=2" */
+void RPCConsole::wallet_zaptxes2()
+{
+    build_parameter_list(QString("-zapwallettxes=2"));
+}
+
+/** Restart wallet with "-upgradewallet" */
+void RPCConsole::wallet_upgrade()
+{
+    build_parameter_list(QString("-upgradewallet"));
+}
+
+/** Restart wallet with "-reindex" */
+void RPCConsole::wallet_reindex()
+{
+    build_parameter_list(QString("-reindex"));
+}
+
+/** Build command-line parameter list for restart */
+void RPCConsole::build_parameter_list(QString arg)
+{
+    // Get command-line arguments and remove the application name
+    QStringList args = QApplication::arguments();
+    args.removeFirst();
+    // Append repair parameter to command line. We don't care whether it might already be there
```

Comment:
Hmmm... It could be very confusing when you use it several times in a row. For example you run wallet and click _zapwallet_, it restarts, zaps it all - all good. But then you click _rescan_ and wallet restarts again and... starts zapping... There is no info presented to the user on that behavior and nothing restricts him from doing so. I propose to at least warn user or clean old "salvage", "rescan", "zap", "upgrade", "reindex" args here. The latter one is a better way imo.